cogito/put.go

```diff
@@ -277,14 +278,32 @@ type gitURL struct {
 	Repo  string
 }
 
+// safeParseUrl parse the given URL and return an error if the URL is not well formated.
+// this avoid to eventually leak passwords in case of error
+//
+// From https://github.com/golang/go/issues/53993
+func safeParse(rawURL string) (*url.URL, error) {
```

Comment:
Suggest to rename to `safeUrlParse` to mimic the name of the wrapped function, `url.Parse`.